### PR TITLE
fix(nemesis): mgmt restore timeout

### PR DESCRIPTION
**Pull request by fgelcer**
fix(nemesis): mgmt restore timeout

since #6733 we made changes to this nemesis
(because of OSS vs enterprise system under test
and snapshot versions), but the timeout is still
broken, and from the staging jobs, they did
not fully pass.
this last fix is to align the use of the
`wait_for` calls, where step and timeout
should work now.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2818,16 +2818,15 @@ def execute_data_validation_thread(command_template, keyspace_name, number_of_ro
             self.log.info("Restoring the schema of the keyspace '%s'", chosen_snapshot_info["keyspace_name"])
             restore_task = mgr_cluster.create_restore_task(restore_schema=True, location_list=location_list,
                                                            snapshot_tag=chosen_snapshot_tag)
-            step = 10
-            extra_time = 300
-            restore_task.wait_and_get_final_status(step=step,
-                                                   timeout=chosen_snapshot_info['expected_timeout'] / step + extra_time)
+
+            restore_task.wait_and_get_final_status(step=10, timeout=chosen_snapshot_info['expected_timeout'])
```
Comment (fruch) [inline sdcm/nemesis.py:2822]: why can't we just revert this change ? to the static timeout it had ? 

this shouldn't take 30min and even not 15min, it's just the system tables 
if it's stuck we have a bug somewhere, and should be handled, not waiting more.

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -2818,16 +2818,15 @@ def execute_data_validation_thread(command_template, keyspace_name, number_of_ro
...
             assert restore_task.status == TaskStatus.DONE, f'Schema restoration of {chosen_snapshot_tag} has failed!'
             self.cluster.restart_scylla()  # After schema restoration, you should restart the nodes
 
         restore_task = mgr_cluster.create_restore_task(restore_data=True,
                                                        location_list=location_list, snapshot_tag=chosen_snapshot_tag)
-        restore_task.wait_and_get_final_status(step=30, timeout=chosen_snapshot_info["expected_timeout"])
+        timeout_plus_extra_time = 1.3 * chosen_snapshot_info['expected_timeout']
```
Comment (fruch) [inline sdcm/nemesis.py:2828]: where is 1.3 coming from ?

File: sdcm/nemesis.py (hunk 1 above)
Comment (fgelcer) [reply sdcm/nemesis.py:2822]: because it is not being enough... and the case i'm currently testing (the most challenging one) is this nemesis during prepare (in the individual nemesis), so it won't be enough 120 seconds

File: sdcm/nemesis.py (hunk 2 above)
Comment (fgelcer) [reply sdcm/nemesis.py:2828]: i'm giving it a tolerance to timeout to another 30%

Comment (fgelcer) [reply sdcm/nemesis.py:2828]: 1 out of 4 nemesis runs passed [here](https://argus.scylladb.com/test/37bfd989-07ba-4778-8899-36d02a013aa9/runs?additionalRuns[]=fb4c19af-675c-4717-b0a2-346fac44c5b6)

still need some work, or to cancel in this case nemesis during prepare (will test that too)

Comment (fruch) [reply sdcm/nemesis.py:2828]: > i'm giving it a tolerance to timeout to another 30%

If we need extra time, why not update the source yaml with this extra time ?

File: sdcm/nemesis.py (hunk 1 above)
Comment (fruch) [reply sdcm/nemesis.py:2822]: The expected timeouts are those measured for the actual restore aren't they ?

It's a bit weird to expect the restore of the scheme to take as long the actual data restore, maybe 2min isn't enough and we need to enlarge it by to 10min, but it should be based on the number that were defined for the data

Comment (fgelcer) [reply sdcm/nemesis.py:2822]: again, my suspicious is about these nemesis runs during prepare taking much much longer than initially planned

**Comment by fgelcer** [comment]
[here it worked](https://argus.scylladb.com/test/37bfd989-07ba-4778-8899-36d02a013aa9/runs?additionalRuns[]=4e2d690c-a765-4697-b040-703cc66207cd) (and adjusted the timeouts to be closer to the reality of these results), although it had a failure, that i will investigate

**Comment by fgelcer** [comment]
@rayakurl @karol-kokoszka @fruch , please review so we can merge this one, that is affecting all restore nemesis scenarios...
i discussed that with Shlomo, and the problem seems to be that in the manager 3.2, we run a repair operation as part of the restore, making the "restore" operation take much longer

**Review by fruch — APPROVED** [review]
Lgtm
